plugins/kafka-info/src/components/KafkaInfoComponent/KafkaInfoComponent.tsx

```diff
@@ -27,49 +103,72 @@ export function KafkaInfoComponent() {
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(false);
 
+  // Pull the metric we're going to use to track our lag
+  const [lagMetric, _setLagMetric] = useState<string>(config.getString('kafkaInfo.lagMetric'));
+
   // Set up some state info for the response from the backend
   const [metricResponse, setMetricResponse] = useState<Object>({});
   const [filteredResponse, setFilteredResponse] = useState<Object>({});
 
+
+  // Get cluster config
+  const [clusterMap, _setClusterMap] = useState(config.get('kafkaInfo.clusters'));
+  const [currentClusterIdx, setCurrentClusterIdx] = useState(0);
+
   // Get backend URL from the config
-  const backendUrl = config.getString('backend.baseUrl');
+  const [backendUrl, _setBackendUrl] = useState(config.getString('backend.baseUrl'));
 
   // Get defined consumer group from entity
   const consumerGroup = entity.metadata.annotations?.[KAFKA_INFO_ANNOTATION].split(',') ?? '';
 
   useEffect(() => {
-    setLoading(true);
-    // Directly query a prometheus endpoint for metric data
-    fetch(`${backendUrl}/api/proxy/kafka-lag/query?query=aws_kafka_max_offset_lag_sum`)
-      .then(response => {
-        return response.json();
-      })
-      .then(text => {
-        setMetricResponse(text);
+    const getClusterAttribute = (attribute: string) => clusterMap[currentClusterIdx]?.[attribute] || '';
+    const fetchLags = (proxy: string) => {
+      setLoading(true);
+      // Directly query a prometheus endpoint for metric data
+      fetch(`${backendUrl}/api/proxy/${proxy}/api/v1/query?query=${lagMetric}`, {
```

Comment:
We need to switch from using vanilla `fetch` to the Backstage provided `fetchApi.fetch` instead so we can make use of authenticated routes.

See here https://github.com/RedHatInsights/backstage-plugin-visual-qontract/blob/master/plugins/webrca-frontend/src/components/WebRCAFetchComponent/WebRCAFetchComponent.tsx#L104 for an example.